src/WCSimWCDigitizerBase.cc

```diff
@@ -0,0 +1,127 @@
+#include "WCSimWCDigitizerBase.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCDIGITIZERBASE_VERBOSE
+//#define WCSIMWCDIGITIZERBASE_VERBOSE
+#endif
+
+// changed from 940 (april 2005) by MF
+// 960 is unsuitable
+
+//RawSignalHitCollection *collection = new RawSignalHitCollection;
+
+WCSimWCDigitizerBase::WCSimWCDigitizerBase(G4String name,
+					   WCSimDetectorConstruction* myDetector,
+					   WCSimWCDAQMessenger* myMessenger)
+  :G4VDigitizerModule(name)
+{
+  G4String colName = "WCDigitizedStoreCollection";
+  this->myDetector = myDetector;
```

Comment:
move this->myDetector = myDetector; to  myDetector(myDetector) in initializer list